library/src/main/java/com/getstream/sdk/chat/notifications/StreamFirebaseMessagingService.java

```diff
@@ -0,0 +1,20 @@
+package com.getstream.sdk.chat.notifications;
+
+import androidx.annotation.NonNull;
+
+import com.getstream.sdk.chat.StreamChat;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+public class StreamFirebaseMessagingService extends FirebaseMessagingService {
```

Comment:
What do you mean? Did you tell about extra information from beckend or info from another app?
We can add boolean field in data part like "isFromStreamChat" and check this field.
In other way we can implement own MessagingService.

In case other app - we can't receive any messages.

Reply:
what I mean is: what if you use Firebase for Stream Chat *and* something else; will `StreamFirebaseMessagingService` handle messages from both? If yes; how do we make sure that only messages pushed by Stream Chat are handled by this class (or discard non Stream messages asap)?

Reply:
@tbarbugli 
We will receive all messages. If we want to extend this some logic we should create own `FirebaseMessagingService` and implement `onReceiveFirebaseMessage` and `setFirebaseToken`.
If we need to filter others messages we should add some field in data part that we can detect messages from StreamChat or not